### ISSUE TITLE
Order by with nulls

### ISSUE THREAD

**Issue by Mc128k** [ujjwalguptaofficial/JsStore#61]
Hi again. I really hope this time It's not my fault. The system doesn't work properly when ordering records that have null values in the order field. This is the query I'm using:

```
          {
            from: 'Job',
            where: [{
              _getCurrentState: {
                in: ['Send', 'Transit'],
              },
            }],
            order: {
              by: '_getReadableFullTicket',
              type: 'asc',
            },
            limit: 20,
          };
```
![screen shot 2018-07-22 at 01 29 58](https://user-images.githubusercontent.com/3678359/43040838-c70081fa-8d4e-11e8-9fd5-d7f6d5e214dd.png)
and it works because all records have the `_getReadableFullTicket` field defined. But if I instead try to order by another field `__lastStateChange` (mapped to timestamp in the screenshot) it behaves weird.
![screen shot 2018-07-22 at 01 30 49](https://user-images.githubusercontent.com/3678359/43040841-e8e16c80-8d4e-11e8-836c-1f1b18058ac3.png)

Is this normal? In theory the null values should stay after fields that are defined. The ordering here doesn't work at all, even if I reverse the sort order.

**Comment by Mc128k** [comment]
Indexeddb is tricky to export, and idbstudio fails to compile atm. I exported a JSON from my data service [here](https://pastebin.com/VWF6RSEe), it should work with the query. Hope it's useful.

**Comment by ujjwalguptaofficial** [comment]
ok thanks. 
What do you mean by this ?  

> idbstudio fails to compile atm

**Comment by Mc128k** [comment]
![screen shot 2018-07-22 at 14 38 21](https://user-images.githubusercontent.com/3678359/43045645-e5f30e3e-8dbc-11e8-9472-9cbae1858ec1.png)

Also I have seen that you added the "Example" section on the doc website and it has issues with Safari (chrome is ok)
![screen shot 2018-07-22 at 14 39 25](https://user-images.githubusercontent.com/3678359/43045662-15ce2918-8dbd-11e8-86ff-b1c2fdbec472.png)

**Comment by ujjwalguptaofficial** [comment]
Ok cool. 

Type of "_getReadableFullTicket" is string and that's why you are getting unexpected result. See screenshot -
![image](https://user-images.githubusercontent.com/10938615/43046115-5cd37eba-8de2-11e8-8652-03fd47983929.png)
 
What do you think ?

**Comment by Mc128k** [comment]
Actually the query that orders by readable full ticket works! It's the one that orders with `__lastStateChange` (which has numbers and nulls) that fails

**Comment by ujjwalguptaofficial** [comment]
Can you give me the whole query ?

**Comment by Mc128k** [comment]
Sure (removed some states before for clarity)
```
{
            from: 'Job',
            where: [{
              _getCurrentState: {
                in: ['Send', 'Transit', 'WaitStart', 'Diagnostics', 'ExternalLab', 'WaitApproval', 'WaitContinue',
                       'Working', 'WaitParts', 'FinalTest'],
              },
            }],
            order: {
              by: '__lastStateChange',
              type: 'asc',
            },
            limit: 20,
          };
```

same error with or without `limit`

**Comment by ujjwalguptaofficial** [comment]
Thanks John. One more thing - could you please give me your table schema 😄, I see there are many columns , and its really pain & time taking to write the table schema. So please 😸

**Comment by ujjwalguptaofficial** [comment]
I mean jsstore schema. Like this 

```
var table = {
        name: 'Shippers',
        columns: [{
                name: "ShipperID",
                primaryKey: true,
                autoIncrement: true
            },
            {
                name: "ShipperName",
                notNull: true,
                dataType: 'string'
            },
            {
                name: "Phone",
                notNull: true,
                dataType: 'string'
            }
        ]
    }
```

It will also help me to find any more issue.  Let me know, if there is any problem i will write by myself.

Thanks

**Comment by Mc128k** [comment]
https://pastebin.com/MYLacTzr

**Comment by ujjwalguptaofficial** [comment]
@Mc128k - the issue is fixed in V 2.2.4. Please check and let me know.
